submissions/Valkirin/frontend-nanodegree-arcade-game/js/app.js

```diff
@@ -1,80 +1,85 @@
-var Enemy = function(x, y, speed) {
-    this.x = x;
-	this.y = y;
-	this.speed = speed;
-    this.sprite = 'images/enemy-bug.png';
+var enemySize = 60;
+var canvasWidth = 550;
+var borgSpawn = -100;
+var playerSpawnX = 200;
+var playerSpawnY = 400;
+var horizontalStep = 100;
+var verticalStep = 83;
+var stepZone = 400;
+
+var Enemy = function(x, y, speed, player) {
+  this.x = x;
+  this.y = y;
+  this.speed = speed;
+  this.sprite = 'images/enemy-bug.png';
+  this.player = player;
 };
 
 Enemy.prototype.update = function(dt) {
-    this.x +=  + (this.speed * dt);
-    
-    if (this.x > 550) {
-    	this.x = -100;
-		this.speed = Math.floor(100 + (Math.random() * 200));
-    };
-    
-    if (player.x > this.x - 60 && 
-        player.x < this.x + 60 && 
-        player.y > this.y - 60 &&
-        player.y < this.y + 60) {
-        player.x = 200;
-        player.y = 400;
-	};
-    
+  this.x += +(this.speed * dt);
+
+  if (this.x > canvasWidth) {
+    this.x = borgSpawn;
+    this.speed = Math.floor(100 + Math.random() * 200);
+  }
+
+  if (
+    player.x > this.x - enemySize &&
+    player.x < this.x + enemySize &&
+    player.y > this.y - enemySize &&
+    player.y < this.y + enemySize
+  ) {
+    player.x = playerSpawnX;
+    player.y = playerSpawnY;
+  }
 };
 
 Enemy.prototype.render = function() {
-    ctx.drawImage(Resources.get(this.sprite), this.x, this.y);
+  ctx.drawImage(Resources.get(this.sprite), this.x, this.y);
 };
 
-var allEnemies = []
-var enemyLocation = [60, 140, 220]
+var enemyLocation = [60, 140, 220];
 
-enemyLocation.forEach(function (y) {
-    allEnemies.push(new Enemy(0, y, 300))
-});
+var allEnemies = enemyLocation.map(y => new Enemy(0, y, 300, player));
 
-// Now write your own player class
-var Player = function(x, y){
-	this.x = x;
-	this.y = y;
-	this.sprite = 'images/char-boy.png';
-}
+var Player = function(x, y) {
+  this.x = x;
+  this.y = y;
+  this.sprite = 'images/char-boy.png';
+};
 
-Player.prototype.update = function(){
-	if (this.y < 60) {
-		this.x = 200;
-		this.y = 400;
-	};
-	
-}
+Player.prototype.update = function() {
+  if (this.y < 60) {
```

Comment:
60, 140, 200, 300 still look mysterious.
However, it's up to you if you fix those.